client/my-sites/sidebar-unified/menu.jsx

```diff
@@ -20,15 +21,26 @@ import { toggleMySitesSidebarSection as toggleSection } from 'state/my-sites/sid
 import ExpandableSidebarMenu from 'layout/sidebar/expandable';
 import MySitesSidebarUnifiedItem from './item';
 import SidebarCustomIcon from 'layout/sidebar/custom-icon';
+import { isExternal } from 'lib/url';
 
-export const MySitesSidebarUnifiedMenu = ( { slug, title, icon, children, path } ) => {
+export const MySitesSidebarUnifiedMenu = ( { slug, title, icon, children, path, link } ) => {
 	const reduxDispatch = useDispatch();
 	const sectionId = 'SIDEBAR_SECTION_' + slug;
 	const isExpanded = useSelector( ( state ) => isSidebarSectionOpen( state, sectionId ) );
 
 	return (
 		<ExpandableSidebarMenu
-			onClick={ () => reduxDispatch( toggleSection( sectionId ) ) }
+			onClick={ () => {
+				if ( link ) {
+					if ( isExternal( link ) ) {
+						// If the URL is external, page() will fail to replace state between different domains
+						document.location.href = link;
```

Comment:
Nit: We have a `externalRedirect()` helper as part of `lib/route/path`, perhaps we could have used that?

Reply:
Thanks for the heads up! I opened https://github.com/Automattic/wp-calypso/pull/46123